.github/scripts/install-slices

```diff
@@ -0,0 +1,324 @@
+    """
+    Minimal data class to store package info.
+    """
+
+    path: str
+    package: str
+    slices: list[str]
+
+    def full_slice_name(self, slice: str) -> str:
+        return f"{self.package}_{slice}"
+
+
+def parse_package(filepath: str) -> Package:
+    """
+    Parse a slice definition file and return the Package.
+    """
+    logging.debug(f"Parsing {filepath}...")
+    with open(filepath, "r") as stream:
+        try:
+            data = yaml.safe_load(stream)
+        except yaml.YAMLError as e:
+            logging.error(f"{filepath}: {e}")
+            exit(1)
+    p = Package()
+    p.path = filepath
+    try:
+        p.package = data["package"]
+        p.slices = []
+        for key in data["slices"]:
+            p.slices.append(key)
+        p.slices = sorted(p.slices)
+    except KeyError as e:
+        logging.error(f"{filepath}: key {e} not found")
+        exit(1)
+    except Exception as e:
+        logging.error(f"{filepath}: {e}")
+        exit(1)
+    return p
+
+
+def query_package_existence(
+    packages: list[str],
+    arch: list[str] = [],
+    components: list[str] = [],
+    suites: list[str] = [],
+) -> tuple[list[str], list[str]]:
+    """
+    Check which packages exist in the archive. Return a list of packages
+    that exist and another list for which do not.
+    """
+    # prepare cmd
+    args = ["rmadison"]
+    if len(arch) > 0:
+        args += ["--architecture", ",".join(arch)]
+    if len(components) > 0:
+        args += ["--component", ",".join(components)]
+    if len(suites) > 0:
+        args += ["--suite", ",".join(suites)]
+    args.append(" ".join(packages))
+    # query the archives using rmadison
+    logging.debug("Querying the archives for packages...")
+    logging.debug("Executing " + " ".join(args))
+    res = subprocess.run(args, capture_output=True, text=True)
+    if res.returncode != 0:
+        logging.error(f"Failed to query the archives {res.returncode}")
+        exit(res.returncode)
+    output = res.stdout.rstrip()
+    logging.debug(f"Archive query output:\n{output}")
+    # parse the output for available packages
+    available = {}
+    for line in output.split("\n"):
+        line = line.strip()
+        if line == "":
+            continue
+        pkg = line.split("|")[0].strip()
+        available[pkg] = True
+    found = []
+    missing = []
+    for p in packages:
+        if available.get(p):
+            found.append(p)
+        else:
+            missing.append(p)
+    return found, missing
+
+
+def ensure_package_existence(packages: list[str], archive: Archive) -> None:
+    """
+    Ensure that packages exist in the archive for any arch.
+    """
+    _, missing = query_package_existence(
+        packages,
+        components=archive.components,
+        suites=archive.suites,
+    )
+    if len(missing) > 0:
+        logging.error(
+            "The following packages do not exist:\n"
+            + "\n".join(f"  - {p}" for p in missing)
```

Comment:
The error message by itself isn't too enlightening cause it's missing at least the `release`. This is probably a good place to use `archive.version`

Reply:
Done, thanks!